### PR TITLE
Remove OpenShift leader-election config

### PR DESCRIPTION
**Pull request by qbarrand**
This OpenShift-specific configuration does not belong upstream.

/cc @yevgeny-shnaidman @pacevedom

### CODE REVIEW

**Review by pacevedom — APPROVED** [review]

File: main.go [hunk 1]
```diff
@@ -71,16 +70,13 @@ func main() {
 	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
 
 	opts := &ctrl.Options{
+		LeaderElection:     cl.EnableLeaderElection,
```
Comment (yevgeny-shnaidman) [inline main.go:73]: what about configuration of the Leaseduration, RetryPeriod etc'? We are going to use the defaults?

Comment (qbarrand) [reply main.go:73]: Upstream yes, we only started specifying those to match what OpenShift operators generally do.

**Pull request merged**